docker/preparation/reformat_db.R

```diff
@@ -0,0 +1,265 @@
+#'
+#' @param algo_id Algorithm id (eg. 'mlr.classif.ranger')
+#' @param task_id A single task_id, on which the algorithm has been run.
+#' @param parameter_names A vector or list of 
+#'
+#' @return A dataframe containing: A column "setup", with the setup_id. A column "<parameter_name>" for every parameter. And one row of data for every setup, that has been run with one of the given algorithms, containing the parameter_data of that run.
+getTableFromDB = function(task_id, algo_id) {
+  parameter_names = names(parameter_ranges[[algo_id]]$pars)
+  
+  db_entries = lapply(parameter_names, function(parameter_name) {
+    sql.exp = paste0("SELECT DISTINCT run.rid, input_setting.setup, input_setting.value AS `", parameter_name, "`
+                     FROM ",mysql_dbname_from, ".input
+                     JOIN ",mysql_dbname_from, ".input_setting ON input_setting.input_id = input.id
+                     JOIN ",mysql_dbname_from, ".run ON run.setup = input_setting.setup
+                     JOIN ",mysql_dbname_from, ".implementation ON input.implementation_id = implementation.id
+                     WHERE input.name = '", parameter_name,"'
+                     AND task_id = ", task_id, "
+                     AND implementation.name = 'mlr.", algo_id, "'")
+    result = dbGetQuery(con, sql.exp)
+    for (i in seq_len(ncol(result))) {
+      if (is.character(result[[i]])) {
+        result[[i]] = type.convert(result[[i]])  
+      }
+    }
+    return(result)
+  })
+  
+  if(length(db_entries) == 0) {
+    return(NULL)
+  }
+  
+  t = db_entries[[1]]
+  # Merge results by same setup_ids
+  for (i in seq_along(db_entries)[-1]) {
+    t = merge(t, db_entries[[i]], all = TRUE, by = c("rid", "setup"))
+  }
+  
+  setDT(t)
+  
+  # Add evaluation measures from `evaluation`-table.
+  if(nrow(t) > 0) {
+    function_ids = c(4,45,54,59,63)
+    method_names = c("auc","accuracy","rmse","scimark","runtime")
+    
+    run_ids = sprintf("(%s)", paste0(simplify2array(t[,"rid"]), collapse=", "))
+    
+    measures = sprintf("SELECT `source` AS `rid`, `function_id`, `value` FROM %s.`evaluation` WHERE `source` IN %s AND `function_id` IN (4,45,54,59,63)", mysql_dbname_from, run_ids)
+    result = dbGetQuery(con, measures)
+  
+    # Merge evaluation measures one-by-one into the table.
+    for (i in seq_along(function_ids)) {
+      function_id = function_ids[i]
+      method_name = method_names[i]
+      t2 = result[result[["function_id"]] == function_id, c("rid","value")]
+      t3 = data.table(rid = t2$rid)
+      t3[[method_name]] = t2$value
+      t = merge(t, t3, all = TRUE, by = "rid")
+    }
+  }
+
+  return(t)
+}
+
+#' Return a list of parameter definitions. This list contains every necessary parameter for the given algorithm.
+#'
+#' @param task_id This is `task_id` from the table `run` in the database.
+#'
+#' @return A named list with one entry for each needed parameter.
+get_params_for_algo = function(algo_name) {
```

Comment:
This function basically just takes `parameter_ranges[[algo_name]]` and converts it to a list that can be serialized to JSON. There should be no reason to call it if the serialization will not be applied later on. You can just write `parameter_ranges[[algo_name]]` instead.